File: 153. Find Minimum in Rotated Sorted Array/Find Minimum in Rotated Sorted Array.cpp
#include <climits>
#include <vector>
using namespace std;

class Solution {
public:
    int findMin(vector<int>& nums) {
        int l = 0;
        int r = nums.size()-1;
        int mNum = INT_MAX;

        while (l <= r){
            int mid = l+(r-l)/2;

            if (nums[r]<nums[mid]){
                l=mid+1;
            }
            else{
                r=mid-1;
            }
            mNum=min(mNum,nums[mid]);
        }
        return mNum;
        
    }
};

File: First 100 completed solutions/153. Find Minimum in Rotated Sorted Array/Find Minimum in Rotated Sorted Array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        l=0
        r=len(nums)-1
        mNum=float('inf')

        while l<=r:
            mid=l+(r-l)//2
            if nums[r]<nums[mid]:
                l=mid+1
            else:
                r=mid-1
            mNum=min(nums[mid],mNum)
        return mNum
        